From 5f7595454a169ca33974db4138bb2e1e6724c243 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Wed, 9 Mar 2022 17:13:01 +0300
Subject: [PATCH 032/151] arm64: dts: renesas: r8a779f0-vc4-ctrl-domain.dts:
 deduplicate

Include r8a779f0-vc4.dtds instead of copying it's content

Also remove explicit enable of gpio nodes - those are enabled by default
in r8a779f0-ctrl-domain.dtsi file.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 .../dts/renesas/r8a779f0-vc4-ctrl-domain.dts  | 261 +-----------------
 1 file changed, 1 insertion(+), 260 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-vc4-ctrl-domain.dts b/arch/arm64/boot/dts/renesas/r8a779f0-vc4-ctrl-domain.dts
index d087d9dd6608..2c1d4dc693ec 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-vc4-ctrl-domain.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-vc4-ctrl-domain.dts
@@ -8,264 +8,5 @@
  * Copyright (C) 2021 Renesas Electronics Corp.
  */
 
-/dts-v1/;
-
-#include "r8a779f0.dtsi"
+#include "r8a779f0-vc4.dts"
 #include "r8a779f0-ctrl-domain.dtsi"
-#include <dt-bindings/gpio/gpio.h>
-
-/ {
-	model = "Renesas VC4 board based on r8a779f0";
-	compatible = "renesas,vc4", "renesas,r8a779f0";
-
-	aliases {
-		serial0 = &scif3;
-		serial1 = &scif0;
-		eth0 = &rswitch;
-	};
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x78000000>;
-	};
-
-	reg_1p8v: regulator-1p8v {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-1.8V";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	reg_3p3v: regulator-3p3v {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-3.3V";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	vcc_sdhi: regulator-vcc-sdhi {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpio = <&gpio1 24 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-};
-
-&extal_clk {
-	clock-frequency = <20000000>;
-};
-
-&extalr_clk {
-	clock-frequency = <32768>;
-};
-
-&gpio4 {
-	status = "okay";
-};
-
-&gpio5 {
-	status = "okay";
-};
-
-&gpio6 {
-	status = "okay";
-};
-
-&gpio7 {
-	status = "okay";
-};
-
-&i2c1 {
-	status = "okay";
-
-	pinctrl-0 = <&i2c1_pins>;
-	pinctrl-names = "default";
-
-	clock-frequency = <400000>;
-
-	rtc@6f {
-		compatible = "isil,isl1208";
-		reg = <0x6f>;
-	};
-};
-
-&i2c4 {
-	status = "okay";
-
-	pinctrl-0 = <&i2c4_pins>;
-	pinctrl-names = "default";
-
-	clock-frequency = <400000>;
-
-	eeprom@50 {
-		compatible = "st,24c16", "atmel,24c16";
-		reg = <0x50>;
-
-		vcc-supply = <&reg_3p3v>;
-
-		label = "board-data";
-		pagesize = <8>;
-	};
-
-	fpga4000@71 {
-		compatible = "renesas,vc4-fpga4000";
-		reg = <0x71>;
-	};
-};
-
-&mmc0 {
-	status = "okay";
-
-	pinctrl-0 = <&sd_pins>;
-	pinctrl-1 = <&sd_pins>;
-	pinctrl-names = "default", "state_uhs";
-
-	vmmc-supply = <&vcc_sdhi>;
-	vqmmc-supply = <&vcc_sdhi>;
-	cd-gpios = <&gpio1 23 GPIO_ACTIVE_LOW>;
-	bus-width = <4>;
-};
-
-&pfc {
-	i2c1_pins: i2c1 {
-		groups = "i2c1";
-		function = "i2c1";
-	};
-
-	i2c4_pins: i2c4 {
-		groups = "i2c4";
-		function = "i2c4";
-	};
-
-	pcie0_pins: pcie0 {
-		groups = "pcie0_clkreq_n";
-		function = "pcie";
-	};
-
-	pcie1_pins: pcie1 {
-		groups = "pcie1_clkreq_n";
-		function = "pcie";
-	};
-
-	scif0_pins: scif0 {
-		groups = "scif0_data", "scif0_ctrl";
-		function = "scif0";
-	};
-
-	scif3_pins: scif3 {
-		groups = "scif3_data", "scif3_ctrl";
-		function = "scif3";
-	};
-
-	scif_clk_pins: scif_clk {
-		groups = "scif_clk";
-		function = "scif_clk";
-	};
-
-	sd_pins: sd {
-		groups = "mmc_data4", "mmc_ctrl";
-		function = "mmc";
-		power-source = <3300>;
-	};
-
-	tsn0_pins: tsn0 {
-		mux {
-			groups = "tsn0_link", "tsn0_mdio";
-			function = "tsn0";
-			drive-strength = <12>;
-		};
-
-		pins_mdio {
-			groups = "tsn0_mdio";
-			drive-strength = <12>;
-		};
-	};
-};
-
-&pciec0 {
-	status = "okay";
-
-	pinctrl-0 = <&pcie0_pins>;
-	pinctrl-names = "default";
-};
-
-&pciec1 {
-	// Exists on the board, but enabling it currently causes boot hang
-	//status = "okay";
-
-	pinctrl-0 = <&pcie1_pins>;
-	pinctrl-names = "default";
-};
-
-&pcie_bus_clk {
-	clock-frequency = <100000000>;
-};
-
-&rswitch {
-	status = "okay";
-
-	ports {
-		port@0 {
-			phy-mode = "sgmii";
-			phy-handle = <&etha0>;
-			etha0: ethernet-phy@0 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-		port@1 {
-			phy-mode = "sgmii";
-			phy-handle = <&etha1>;
-			etha1: ethernet-phy@1 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-		port@2 {
-			phy-mode = "sgmii";
-			phy-handle = <&etha2>;
-			etha2: ethernet-phy@2 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-	};
-};
-
-&scif0 {
-	status = "okay";
-
-	pinctrl-0 = <&scif0_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-};
-
-&scif3 {
-	status = "okay";
-
-	pinctrl-0 = <&scif3_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-};
-
-&scif_clk {
-	pinctrl-0 = <&scif_clk_pins>;
-	pinctrl-names = "default";
-
-	clock-frequency = <24000000>;
-};
-- 
2.17.1

